subsys/testsuite/coverage/coverage.c

```diff
@@ -57,18 +57,12 @@ static inline void print_u32(uint32_t v)
 }
 
 /**
- * write_u64 - Store 64 bit data on a buffer and return the size
+ * print_u64 - Print 64 bit of gcov data
  */
-
-static inline void write_u64(void *buffer, size_t *off, uint64_t v)
+static inline void print_u64(uint64_t v)
 {
-	if (buffer != NULL) {
-		memcpy((uint8_t *)buffer + *off, (uint8_t *)&v, sizeof(v));
```

Comment:
this `memcpy()` was actually not executed

Reply:
How did you get to this conclusion?

Reply:
lets continue this discussion in one place - in #64396

Reply:
> this `memcpy()` was actually not executed

sorry, this my statement was not clear, it should be "optimized out" instead - `static inline void write_u64()` has an inline assembler commands instead memcpy(), so gcovr has no function call to match at this place. The proposed change has a memcpy() call in `gcov_to_gcda()`

```
Dump of assembler code for function write_u64:
   0x00108a9b <+0>:     push   %ebp
   0x00108a9c <+1>:     mov    %esp,%ebp
   0x00108a9e <+3>:     sub    $0x18,%esp
   0x00108aa1 <+6>:     mov    0x10(%ebp),%eax
   0x00108aa4 <+9>:     mov    %eax,-0x10(%ebp)
   0x00108aa7 <+12>:    mov    0x14(%ebp),%eax
   0x00108aaa <+15>:    mov    %eax,-0xc(%ebp)
   0x00108aad <+18>:    mov    0x11e240,%eax
   0x00108ab2 <+23>:    mov    0x11e244,%edx
   0x00108ab8 <+29>:    add    $0x1,%eax
   0x00108abb <+32>:    adc    $0x0,%edx
   0x00108abe <+35>:    mov    %eax,0x11e240
   0x00108ac3 <+40>:    mov    %edx,0x11e244
   0x00108ac9 <+46>:    cmpl   $0x0,0x8(%ebp)
   0x00108acd <+50>:    je     0x108b03 <write_u64+104>
-------------------------------------------------------------- memcpy()
   0x00108acf <+52>:    mov    0x11e248,%eax
   0x00108ad4 <+57>:    mov    0x11e24c,%edx
   0x00108ada <+63>:    add    $0x1,%eax
   0x00108add <+66>:    adc    $0x0,%edx
   0x00108ae0 <+69>:    mov    %eax,0x11e248
   0x00108ae5 <+74>:    mov    %edx,0x11e24c
   0x00108aeb <+80>:    mov    0xc(%ebp),%eax
   0x00108aee <+83>:    mov    (%eax),%edx
   0x00108af0 <+85>:    mov    0x8(%ebp),%eax
   0x00108af3 <+88>:    lea    (%edx,%eax,1),%ecx
   0x00108af6 <+91>:    mov    -0x10(%ebp),%eax
   0x00108af9 <+94>:    mov    -0xc(%ebp),%edx
   0x00108afc <+97>:    mov    %eax,(%ecx)
   0x00108afe <+99>:    mov    %edx,0x4(%ecx)
   0x00108b01 <+102>:   jmp    0x108b60 <write_u64+197>
--------------------------------------------------------------------------
   0x00108b03 <+104>:   lea    -0x10(%ebp),%eax
   0x00108b06 <+107>:   mov    (%eax),%eax
   0x00108b08 <+109>:   sub    $0xc,%esp
   0x00108b0b <+112>:   push   %eax
   0x00108b0c <+113>:   call   0x10899f <print_u32>
   0x00108b11 <+118>:   add    $0x10,%esp
   0x00108b14 <+121>:   mov    0x11e250,%eax
   0x00108b19 <+126>:   mov    0x11e254,%edx
   0x00108b1f <+132>:   add    $0x1,%eax
   0x00108b22 <+135>:   adc    $0x0,%edx
   0x00108b25 <+138>:   mov    %eax,0x11e250
   0x00108b2a <+143>:   mov    %edx,0x11e254
   0x00108b30 <+149>:   lea    -0x10(%ebp),%eax
   0x00108b33 <+152>:   add    $0x4,%eax
   0x00108b36 <+155>:   mov    (%eax),%eax
   0x00108b38 <+157>:   sub    $0xc,%esp
   0x00108b3b <+160>:   push   %eax
   0x00108b3c <+161>:   call   0x10899f <print_u32>
   0x00108b41 <+166>:   add    $0x10,%esp
   0x00108b44 <+169>:   mov    0x11e258,%eax
   0x00108b49 <+174>:   mov    0x11e25c,%edx
   0x00108b4f <+180>:   add    $0x1,%eax
   0x00108b52 <+183>:   adc    $0x0,%edx
   0x00108b55 <+186>:   mov    %eax,0x11e258
   0x00108b5a <+191>:   mov    %edx,0x11e25c
   0x00108b60 <+197>:   mov    0xc(%ebp),%eax
   0x00108b63 <+200>:   mov    (%eax),%eax
   0x00108b65 <+202>:   lea    0x8(%eax),%edx
   0x00108b68 <+205>:   mov    0xc(%ebp),%eax
   0x00108b6b <+208>:   mov    %edx,(%eax)
   0x00108b6d <+210>:   nop
   0x00108b6e <+211>:   mov    %ebp,%esp
   0x00108b70 <+213>:   pop    %ebp
   0x00108b71 <+214>:   ret
```